app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -2018,6 +2021,63 @@ class ChatActivity :
             }
     }
 
+    @SuppressLint("LongLogTag")
+    private fun saveImageToStorage(
+        message: ChatMessage,
+        openWhenDownloaded: Boolean,
+        funToCallWhenDownloadSuccessful: (() -> Unit)
+    ) {
+        message.openWhenDownloaded = openWhenDownloaded
+        adapter?.update(message)
+
+        val fileName = message.selectedIndividualHashMap!!["name"]
+        var size = message.selectedIndividualHashMap!!["size"]
+        val sourceFilePath = applicationContext.cacheDir.path
+
+        if (size == null) {
+            size = "-1"
+        }
+        val fileSize = size.toLong()
+        val fileId = message.selectedIndividualHashMap!!["id"]
+
+        // check if download worker is already running
+        val workers = WorkManager.getInstance(
+            context
+        ).getWorkInfosByTag(fileId!!)
+        try {
+            for (workInfo in workers.get()) {
+                if (workInfo.state == WorkInfo.State.RUNNING || workInfo.state == WorkInfo.State.ENQUEUED) {
+                    Log.d(TAG, "Download worker for $fileId is already running or scheduled")
```

Comment:
just replace "Download Worker" with SaveFileToStorageWorker, so the log message makes sense again.